pom.xml

```diff
@@ -602,6 +602,13 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.junit.jupiter</groupId>
+      <artifactId>junit-jupiter</artifactId>
```

Comment:
I think `junit-jupiter` includes `junit-jupiter-engine`, so if you add it, you should be able to remove that dependency. See https://junit.org/junit5/docs/current/user-guide/#dependency-metadata-junit-jupiter

Also I see many references to `junit-jupiter-engine` above, I think those can be removed as well.

Reply:
Thanks, I'll look at cleaning that up a bit.